python/pyarrow/tests/test_pandas.py

```diff
@@ -4816,3 +4816,23 @@ def test_unhashable_map_keys_with_pydicts():
         for tup1, tup2 in zip(row1, row2):
             assert np.array_equal(tup1[0], tup2[0])
             assert tup1[1] == tup2[1]
+
+
+def test_column_conversion_for_datetime():
+    # GH-35235
+    # pandas implemented __from_arrow__ for DatetimeTZDtype
+    # https://github.com/pandas-dev/pandas/pull/52201
+    if Version(pd.__version__) <= Version("2.0.1"):
+        pytest.skip("__from_arrow__ added to DatetimeTZDtype in pandas version 2.1.0")
```

Comment:
Do we need to skip the test for older pandas? I would assume that everything (except for the `assert hasattr(pandas_dtype, '__from_arrow__')`) should work the same with older pandas? (essentially this is testing that we _preserve_ this behaviour with newer pandas)

Reply:
Yes, you are correct!
I will put the main part of the test outside this skip and leave `assert hasattr(pandas_dtype, '__from_arrow__')` in it.

Reply:
Or just leave out that check, in the end that's testing an implementation detail of pandas, not something we control